test-shunting-yard.cpp

```diff
@@ -1118,3 +1118,34 @@ TEST_CASE("Exception management") {
   REQUIRE_THROWS(ecalc2.compile("map()['hello']]"));
   REQUIRE_THROWS(ecalc2.compile("map(['hello']]"));
 }
+
+TEST_CASE("Variable UTF8 name support") {
+  TokenMap v1;
+  v1["n_"] = 5; // Normal name
+  v1["a←"] = 5; // Name starting with ascii, but contains UTF8
+  v1["łn"] = 5; // Name starting with UTF8, but also has ascii
+  v1["€"] = 5;  // A single UTF8 character
+  v1["€€æøå€"] = 5; // Multiple UTF8 characters
+  v1["€€€"] = 5; // Multiple UTF8 characters
```

Comment:
We should also test something invalid, e.g. `"\xc3_is_not_unicode"`, and it should throw an exception informing that it was expecting UTF8 characters, but what it actually received was not UTF8.

If we could include the hex representation of the invalid character in the exception it would make the debugging easier, so it is a plus if we can make it.